### ISSUE TITLE
Looking for guidance troubleshooting freeze on start

### ISSUE THREAD

**Issue by qouoq** [spyder-ide/spyder#16996]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [ ] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [x] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

Spyder freezes immediately after starting. All widgets load (Editor, Console, Help, other tabs are visible), cursor blinks in the editor, then after maybe 1 or 2 seconds the application doesn't respond (and cursor stops blinking).
The widgets are actually processing keystrokes during the 1 or 2 seconds before freeze (so if I do quickly I can send input to the editor, the console, and also switch to the history tab which shows it is loaded).

### What steps reproduce the problem?

`pacman -S spyder` for the Arch package version, or `git clone` and `bootstrap.py` for the git checkout; I tried only with latest version on 5.x branch, so it's 5.3 for the git checkout while the Arch package is 5.2. Both behave exactly the same, including debug log and gdb backtrace.

### Body of my issue

I know this is not an Anaconda setup and there's no official support, and I would like foremost to ask for guidance in troubleshooting the problem.

What I was able to get so far is a debug log and a backtrace. I did with both Arch 5.2 and git 5.3, from what I can see the result is the same. I assume the results from the git checkout will be most useful:
Spyder log: [debug-checkout.log](https://github.com/spyder-ide/spyder/files/7694792/debug-checkout.log)
gdb py-bt output: [pybt-checkout.log](https://github.com/spyder-ide/spyder/files/7694793/pybt-checkout.log)
gdb raw bt output: [raw-bt-checkout.log](https://github.com/spyder-ide/spyder/files/7694908/raw-bt-checkout.log) _(this is from a different session, but it should match closely, the thread numbers are always the same throughout my tests from what I can see)_

This is from `python bootstrap.py -- --safe-mode --debug-info verbose`. The `py-bt` log shows a `thread apply x-y-z py-bt` followed by `t a x-y-z py-list`, where `x-y-z` is my selection of threads (those that don't say _Unable to locate python frame_).
The Spyder log looks ok to me, I compared with a debug log from a working Spyder on a Windows machine, I don't see any difference, including to the point reached before starting interaction.

From there I don't know how to troubleshoot further. My guess is that I should concentrate on thread 1, but it's no more than a hunch. It spends most of its time in `QApplication::compressEvent` from `libQt5Widgets.so.5`, which may just mean that Qt is gently sitting in its event loop? I think I caught a glimpse of an infinite recursion error while trying to step through, but I'm really not sure.

So, I'd be glad to try to follow any advice for pursuing troubleshooting, which I guess would start with finding more useful logs (possibly installing more debug symbols so that these backtraces look more human)?

### Other notes

* pylsp gets loaded, often with a delay of ~20 seconds (but sometimes it's very quick, couldn't discern a pattern); its status then gets updated in the status bar even though Spyder is not responding. pylsp itself works.
* `jupyter qtconsole` is working beautifully, including "inline help" (not sure what it's called).
* I tried the PySide2 library, it gives the same result, including pretty much the same gdb backtrace.
* I tried to run `runtests.py` from the git checkout; it gave a lot of errors (as well as many successful tests) and eventually crashed with a segmentation fault in test_ipythonconsole.py:
[spyder-runtests.log](https://github.com/spyder-ide/spyder/files/7695002/spyder-runtests.log)

## Versions

* Spyder version: 5.2.0 (Arch package), 5.3.0.dev0 rev 9a0cab850 (git checkout)
* Python version: 3.9.9
* Qt version: 5.15.2
* PyQt version: 5.15.6
* Operating System name/version: Arch Linux

### Dependencies

Cannot get from Spyder, listing the ones that (I'm guessing) could be relevant:
```
IPython: 7.30.1
qtconsole: 5.2.1
qtpy: 1.11.2
spyder-kernels: 2.2.0
lsp-server: 1.3.2
```

**Comment by qouoq** [comment]
Getting some more advice, I tried with another user, and then Spyder is running fine. So the issue is mostly irrelevant.
How can I try to identify the problem with my current user though?

**Comment by qouoq** [comment]
Short of being able to delete the issue, I understand I should pinpoint the user issue. So I'll try to do that and open another issue if needed.
Pleade disregard this issue.

**Comment by satyam-bhardwaj** [comment]
I had the same problem, also on Arch using the official repository package. I ran it under a new user and it worked. In my case I think the issue was kvantum themes. Going back to default Plasma setup resolved the issue on my user as well.

**Comment by qouoq** [comment]
Yes, it's related to kvantum, the way qtconsole (the Spyder console) handles its text area widget leading to a loop of resize events that makes the application unresponsive.
My current workaround is `alias spyder='QT_STYLE_OVERRIDE=Fusion spyder'`

The issue is tracked at #16826.
